### PR TITLE
Allow multiple JavaScript modules on a certain element

### PR DESCRIPTION
**Pull request by alex-ju**
## What
Allow multiple JavaScript modules on a certain element.

## Why
To allow components using scripts from both `govuk-frontend` and `govuk_publishing_components` to work as intended. This fixes a bug where the checkboxes and details components only load the `govuk-frontend` scripts as a result of [converging to a single module system](https://github.com/alphagov/govuk_publishing_components/pull/2095).

This PR proposes a uniform notation between the `data-module` values (e.g. "govuk-checkboxes" or "gem-checkboxes") and the JavaScript module names (`GovukCheckboxes` and `GemCheckboxes`) with the possibility to apply multiple modules on a certain element. This makes the modules system a bit less 'magical' in the sense that its behaviour is more predictable instead of trying to automatically initialise multiple modules based on a single `data-module` declaration.

## Visual Changes
No visual changes.

### CODE REVIEW

File: app/views/govuk_publishing_components/components/_details.html.erb [hunk 1]
```diff
@@ -7,7 +7,7 @@
   css_classes << (shared_helper.get_margin_bottom)
 
   data_attributes ||= {}
-  data_attributes[:module] = 'govuk-details'
+  data_attributes[:module] = 'govuk-details gem-details'
```
Comment (theseanything) [inline app/views/govuk_publishing_components/components/_details.html.erb:10]: Nit: Should this be plural "modules", to indicate you can specify more than one?

**Review by theseanything — APPROVED** [review]
Nice clean up of the module names!!

File: app/views/govuk_publishing_components/components/_details.html.erb (hunk 1 above)
Comment (alex-ju) [reply app/views/govuk_publishing_components/components/_details.html.erb:10]: That's a good point, I'm not 100% confident of the full implications of renaming `data-module` to `data-modules`. Would it be ok with you if I record this as an issue to investigate further?

Comment (theseanything) [reply app/views/govuk_publishing_components/components/_details.html.erb:10]: Sure thing! Not a blocker for this PR.

**Pull request merged**